### PR TITLE
[CodeCamp2023-340] New Version of config Adapting MobileNet Algorithm

### PR DESCRIPTION
**Pull request by DE009**
New Version of config Adapting MobileNet Algorithm, based on https://github.com/open-mmlab/OpenMMLabCamp/discussions/340
## Motivation

MMEngine has introduced a novel configuration mechanism, utilizing which we will perform configuration file adaptation.

## Modification
Add 3 config python files in mmpretrain/configs/_base_/datasets/
Add 2 config python files in mmpretrain/configs/_base_/models/
Add 2 config python files in mmpretrain/configs/_base_/schedules/
Add 1 config python files in mmpretrain/configs/mobilenet_v2/
Add 5 config python files in mmpretrain/configs/mobilenet_v3/
Remove folder mmpretrain/configs/_base_/models/mobilenet_v3/

## Checklist

**Before PR**:

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by OpenMMLab-Assistant-004** [comment]
Hi @DE009,

We'd like to express our appreciation for your valuable contributions to the mmpretrain. Your efforts have significantly aided in enhancing the project's quality.
It is our pleasure to invite you to join our community thorugh Discord_Special Interest Group (SIG) channel. This is a great place to share your experiences, discuss ideas, and connect with other like-minded people. To become a part of the SIG channel, send a message to the moderator, OpenMMLab, briefly introduce yourself and mention your open-source contributions in the #introductions channel. Our team will gladly facilitate your entry. We eagerly await your presence. Please follow this link to join us: ​https://discord.gg/UjgXkPWNqA.

If you're on WeChat, we'd also love for you to join our community there. Just add our assistant using the WeChat ID: openmmlabwx. When sending the friend request, remember to include the remark "mmsig + Github ID".

Thanks again for your awesome contribution, and we're excited to have you as part of our community!

File: mmpretrain/configs/_base_/models/mobilenet_v3/mobilenet_v3_small_imagenet.py
Comment (fangyixiao18) [inline mmpretrain/configs/_base_/models/mobilenet_v3/mobilenet_v3_small_imagenet.py:1]: please remove the folder `_base_/models/mobilenet_v3`, but only keep one file to set model config like `mobilenet_v3_small.py`, and the training configs of mobile net v3 all inherit from this `mobilenet_v3_small.py` file

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

**Comment by fangyixiao18** [comment]
please update these _base_ file and training configs in `mmpretrain/configs/mobilenet_v3`

File: mmpretrain/configs/_base_/models/mobilenet_v3/mobilenet_v3_small_imagenet.py
Comment (fangyixiao18) [reply mmpretrain/configs/_base_/models/mobilenet_v3/mobilenet_v3_small_imagenet.py:1]: you can refer to `mmpretrain/configs/vision_transformer`, to find some examples

Comment (DE009) [reply mmpretrain/configs/_base_/models/mobilenet_v3/mobilenet_v3_small_imagenet.py:1]: removed the folder `_base_/models/mobilenet_v3`
add base model config file `mobilenet_v3_small.py` in `_base_/models/`
modified all training configs of mobile net v3 in folder `mmpretrain/configs/mobilenet_v3/` to inheri the base model config `mobilenet_v3_small.py`

**Review by fangyixiao18 — APPROVED** [review]

**Pull request merged**